### PR TITLE
[LLD] - [LIVE-7618] - Typescript improvements and rework of the language & locale components across LLD.

### PR DESCRIPTION
**Pull request by sshmaxime**
### 📝 Description

**Typescript improvements and rework of the language & locale components across LLD.**

**Additions** of:
- `Language` type.
- `LanguageLocale` type.
- `Locale` type.
- `Locales` type.
- `LanguageDefinition` type.

> There is a bit of a grey space between `Locale` and `LanguageLocale`. In LLD `Locale` and `Language` are 2 completely dissociated elements. As of now `Locale` is represented as a `string` while `LanguageLocale` is a specific type which correspond to the `Locale` that are supported in our Language system. As of now, it's just a way to keep a track of them but it isn't used anywhere, yet.

**DeviceSupport** is directly included in the `Languages` variable.

**Reworked** and typed `useDynamicUrl`. Now if you need an url that requires languages information such as `buyNew` or `genuineCheck` or others you can do: `useDynamicUrl("buyNew")` or `useDynamicUrl("genuineCheck")`.

### ❓ Context

- **Impacted projects**: `LLD` 
- **Linked resource(s)**: [LIVE-7618] 

### ✅ Checklist

- [ ] **Test coverage** 
- [ ] **Atomic delivery** 
- [ ] **No breaking changes** 

### 🚀 Expectations to reach

_Please make sure you follow these [**Important Steps**](https://github.com/LedgerHQ/ledger-live/blob/develop/CONTRIBUTING.md#important-steps)._

_Pull Requests must pass the CI and be internally validated in order to be merged._

[LIVE-7618]: https://ledgerhq.atlassian.net/browse/LIVE-7618?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ofreyssinet-ledger** [comment]
Didn't check the code yet but several remarks on the PR:
- Maybe try to make PR title a bit more explicit so it's easy to find etc.
- Also can you elaborate a bit the description please ? What has been reworked, why etc. Make it exhaustive if possible.
- [You will also need a changeset](https://github.com/LedgerHQ/ledger-live/pull/4251#issuecomment-1669189087) (`pnpm changeset`) so that your changes appear in the changelogs of the next release. Makes it easy to track what is included in a given release in case there is an issue.
- can you elaborate on that please ? 
  > Small breaking change because the language system has been changed.

File: apps/ledger-live-desktop/src/renderer/reducers/settings.ts [hunk 1]
```diff
@@ -100,27 +107,35 @@ export type SettingsState = {
...
+  const systemLocal = getSystemLocale();
+
+  // Find language from system locale (i.e., en, fr, es ...)
+  const languageId = LanguageIds.find(lang => systemLocal.startsWith(lang));
+
+  // If language found, try to find corresponding locale
+  if (languageId) {
+    // const localeId = Languages[languageId].locales.find(lang => systemLocal.startsWith(lang));
+    // TODO Hack because the typing on the commented line above doesn't work
+    const languageLocales = Languages[languageId].locales as Locales;
```
Comment (KVNLS) [inline apps/ledger-live-desktop/src/renderer/reducers/settings.ts:120]: If I define 
`export const Languages: LanguageMap<LanguageDefinition> = {} as const` I can avoid this hack
Could you check if it satisfies what you expect, please?

When I do the change I have relevant errors as follows:
```

 Running typescript type checker on 1657 files…
apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/steps/restore/Language.tsx:29:5 - error TS18048: 'DEFAULT_LANGUAGE.deviceSupport' is possibly 'undefined'.

29     DEFAULT_LANGUAGE.deviceSupport.label,
       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
apps/ledger-live-desktop/src/renderer/screens/settings/sections/General/ChangeDeviceLanguagePromptDrawer.tsx:80:15 - error TS18048: 'DEFAULT_LANGUAGE.deviceSupport' is possibly 'undefined'.

80               DEFAULT_LANGUAGE.deviceSupport.label
                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

That makes sense as we define 
` deviceSupport: { label: DeviceLanguages; id: number } | undefined;`

**Review by KVNLS — CHANGES_REQUESTED** [review]
Great work, you improved our translation management a lot! 🎉
Just a small type detail and you're good to merge.

**Review by ofreyssinet-ledger — CHANGES_REQUESTED** [review]
Very neat, great job! Thanks for taking care of that.
Please generate a changeset so these changes appear in the changelog of the next release 👍

**Review by ofreyssinet-ledger — APPROVED** [review]

**Pull request merged**